src/accelerate/utils/modeling.py

```diff
@@ -233,6 +233,63 @@ def values(self):
         return sum([x[1:] for x in self], [])
 
 
+def check_tied_parameters(model: nn.Module):
+    """
+    Check if there is any indication in the given model that some weights should be tied.
+
+    Args:
+        model (nn.Module): The model to inspect
+
+    Returns:
+        bool: True if the model needs to have tied weights
+    """
+
+    # based on model.tie_weights() method
+    return any(
+        [
+            hasattr(model.config, True)
+            and getattr(model.config, "tie_word_embeddings", False)
+            and model.get_output_embeddings(),
+            hasattr(model.config, True)
+            and getattr(model.config, "is_encoder_decoder", False)
+            and getattr(model.config, "tie_encoder_decoder", False),
+            any(hasattr(module, "_tie_weights") for module in model.modules()),
+        ]
+    )
+
+
+def _get_param_device(param, device_map):
+    if param in device_map:
+        return device_map[param]
+    parent_param = ".".join(param.split(".")[:-1])
+    if parent_param == param:
+        raise ValueError(f"The device_map was do not contain the module {param}.")
+    else:
+        return _get_param_device(parent_param, device_map)
+
+
+def check_tied_parameters_on_same_device(tied_params, device_map):
+    """
+    Check if tied parameters are on the same device
+
+    Args:
+        tied_params (List[List[str]]):  A list of lists of parameter names being all tied together
+        device_map (Dict[str, Union[int, str, torch.device]]): mapping between the parameters and the devices
```

Comment:
Types should be in code blocks. Also copy the docstring of `device_map` from another function.